frontend/public/components/catalog/catalog-page.tsx

```diff
@@ -344,6 +346,30 @@ export const CatalogListPage = withExtensions<CatalogListPageExtensionProps>({
   },
 );
 
+export const filterIncompatibleHelmCharts = (
+  charts: HelmChartEntries,
+  kubernetesVersion: string,
+): HelmChartEntries => {
+  const filteredHelmcharts: HelmChartEntries = _.reduce(
+    charts,
+    (acc, helmCharts: HelmChart[], helmChartName: string) => {
+      let arr: HelmChart[] = [];
+      _.forEach(helmCharts, (obj: HelmChart) => {
+        if (obj?.kubeVersion && semver.valid(kubernetesVersion)) {
+          if (semver.satisfies(kubernetesVersion, obj.kubeVersion)) {
+            arr = [...arr, obj];
+          }
+        } else {
+          arr = [...arr, obj];
```

Comment:
can it be clubbed with above `arr = [ ...arr, obj ]` statement?

Reply:
Use add.push(obj) here as well. I personally like this else case :)

Reply:
@abhinandan13jan  the purpose of adding this in else is if a helm chart doesn't have `kubeVersion` then consider it as compatible